src/app.js

```diff
@@ -8,6 +8,167 @@ import './css/style.css';
+  $detailsElement.html('');
+
+  const generatedHTML = detailsTemplate(tripDetails.attributes);
+  $detailsElement.append(generatedHTML);
+
+  $('#goReservationForm').on('click', function(event) {
+    renderReservationForm(event, tripDetails);
+  });
+};
+
+const renderReservationForm = function renderReservationForm(event, tripDetails){
+  $('#goReservationForm').hide();
+  const tripId = tripDetails.attributes.id
+  const $spaceReservationForm = $('#spaceReservationForm');
+  $spaceReservationForm.html('');
+  const generatedHTML = reservationTemplate({tripId: tripId});
+  // console.log(generatedHTML);
+  $spaceReservationForm.append(generatedHTML);
+  // nuevas cosas
+  event.preventDefault();
+  $('#submitForm').on('click', reserveTripHandler);
+};
+
+const  reserveTripHandler = function reserveTripHandler(event){
+  event.preventDefault();
+  console.log('inside reserveTripHandler');
+  let reservationData = {};
+
+  RESERVATION_FIELDS.forEach((field) =>{
+    const inputElement = $(`#reservation-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    reservationData[field] = value;
+    inputElement.val('');
+  });
+  const reservationEnterUser = new Reservation(reservationData);
+  console.log(`reservationEnterUser ${reservationEnterUser}`);
+  console.log(reservationEnterUser);
+  reservationEnterUser.save({}, { success: (model, response) => {
+    reportStatus('success', 'Make your bags. You reserve this trips Successfully!');
+    console.log("reservation went great");
+  }, error: (model, response) => {
+
+    const errors = response.responseJSON["errors"];
+    for (let field in errors) {
+      for (let problem of errors[field]) {
+        reportStatus('error', `${field}: ${problem}`);
+      }
+    }
+  }
+});
+};
+
+const addTripHandler = function(event) {
+  event.preventDefault();
+
+  const tripData = {};
+  TRIP_FIELDS.forEach((field) => {
+    // select the input corresponding to the field we want
+    const inputElement = $(`#add-trip-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    tripData[field] = value;
+
+    inputElement.val('');
+  });
+  const tripEnterUser =  new Trip(tripData);
+
+  tripEnterUser.save({}, {
+    success: (model, response) => {
+      console.log('Successfully saved trip!');
+      reportStatus('success', 'Successfully saved trip!');
+    },
+    error: (model, response) => {
+      const errors = response.responseJSON["errors"];
+      for (let field in errors) {
+        for (let problem of errors[field]) {
+          reportStatus('error', `${field}: ${problem}`);
+        }
+      }
+    },
+  });
+};
+
+
+let tripTemplate;
+let detailsTemplate;
+let reservationTemplate;
+
+// Clear status messages
+const clearStatus = function clearStatus() {
+  $('.clear').hide();
+  $('#status-messages ul').html('');
+  $('#status-messages').hide();
+};
+
+// Add a new status message
+const reportStatus = function reportStatus(status, message) {
```

Comment:
This function should also clear the container of error messages.  Now they never go away.